### ISSUE TITLE
Missing app DIR

### ISSUE THREAD

**Issue by the-webguys** [ricoberger/go-vue-starter#7]
hello there, maybe i am being stupid, but where is the `cd go-vue-starter/app` DIR it is not in the zip or in the project, but is referenced in the instructions...

**Comment by Bakeelnon** [comment]
I have the same problem, wtf

**Comment by the-webguys** [comment]
if you look in the `\go-vue-starter_old\web\vue.js` DIR you can see the `yarn.lock`. so you can run `yarn install` there, I managed to get that part working.

Since i am on windows, `make build` required some extra work - that is as far as i got before giving up and taking a break

**Comment by Bakeelnon** [comment]
Thanks, that's work. I am on windows too, but I've tried to do make build under the Linux system, and it doesn't work there. I don't know what should I do

**Comment by the-webguys** [comment]
glad that helped - hoping the developer can assist...

is your error like this:
```λ make build
esc -o pkg/static/static.go -pkg static web/vue.js/dist web/mail
process_begin: CreateProcess(NULL, esc -o pkg/static/static.go -pkg static web/vue.js/dist web/mail, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [static] Error 2```

**Comment by Bakeelnon** [comment]
`make` command absolutely doesn't work under my system like `'make' is not recognized as an internal or external command,
operable program or batch file.`
Tomorrow I gonna try fix that

**Comment by the-webguys** [comment]
i did a little google searching and found this: [stack overflow link](https://stackoverflow.com/questions/32127524/how-to-install-and-use-make-in-windows) i installed the **GNUWin32** one, (virus checked it first) tried copying the contents of:  `C:\Program Files (x86)\GnuWin32\bin` over to the `\go-vue-starter` and that is as far as i have got, with that error from above

**Comment by ricoberger** [comment]
Hi I created a PR (#8) to fix the wrong path for the Vue.js app in the readme. I also added the command to install [esc](https://github.com/mjibson/esc) which is required to build the starter.

I havn't a Windows system to assist with `make`. To use the starter without `make` you can use the following commands:

```sh
git clone github.com/ricoberger/go-vue-starter
go get -u github.com/mjibson/esc

# Build the Vue app
cd go-vue-starter/web/vue.js

yarn install
yarn build

# Run the Go server and the Vue app on localhost:8080
cd go-vue-starter

# Build the go application (you can also add the `-ldflags`, but they are not required)
esc -o pkg/static/static.go -pkg static web/vue.js/dist web/mail
go build -o ./bin/starter ./cmd/starter

./bin/starter
```

I hope this helps. If you have any further questions or suggestions for the readme please let me know.

**Comment by the-webguys** [comment]
thank you for this, but erm `./bin/starter` ??

how do i start the server, `./bin/starter` is not a command lol

**Comment by ricoberger** [comment]
Can you try the following on Windows:

```sh
esc -o pkg\static\static.go -pkg static web\vue.js\dist web\mail
go build -o .\bin\starter.exe .\cmd\starter
.\bin\starter.exe
```

**Comment by the-webguys** [comment]
that did something better - thanks :)

```λ .\bin\starter.exe
time="2019-12-17T21:52:13Z" level=info msg="Starting starter (version=, branch=, revision=)"
time="2019-12-17T21:52:13Z" level=info msg="Build context (go=go1.13.5, user=, date=)"
time="2019-12-17T21:52:43Z" level=fatal msg="Could not open database connection" error="server selection timeout"```

couple of questions if i may:
1) do i need to install and setup/configure mongo? not used mongo in some time
2) can i use mysql?
3) what would i need to do to get the email working?

Sorry to be a pain!
If i am having issues, it is likely that others are too..

**Comment by ricoberger** [comment]
Hi @the-webguys, to your questions:

1. MongoDB must be running to use the starter.
2. MySQL is not supported. But you can implement MySQL support in a package like `github.com/ricoberger/go-vue-starter/pkg/db/mysql` where you implement the `DB` interface and adjust the [db.go](https://github.com/ricoberger/go-vue-starter/blob/master/pkg/db/db.go) file to use the MySQL driver:

```go
package db

import (
	"github.com/ricoberger/go-vue-starter/pkg/db/mongodb"
	"github.com/ricoberger/go-vue-starter/pkg/db/mysql"
	"github.com/ricoberger/go-vue-starter/pkg/model"
)

// Config represents the configuration of the database interface
type Config struct {
	MongoDB *mongodb.Config
	MySQL *mysql.Config
}

// DB is the interface which must be implemented by all db drivers
// These are the functions which must be implemented by your MySQL package
type DB interface {
	CloseConnection() error

	CreateUser(u *model.User) error
	GetUser(id string) (*model.User, error)
	GetUserByEmail(email string) (*model.User, error)
	SaveUser(u *model.User) error
	DeleteUser(id string) error
}

// NewConnection creates a new database connection
func NewConnection(config *Config) (DB, error) {
	// Use MySQL
	db, err := mysql.NewConnection(config.MySQL)
	if err != nil {
		return nil, err
	}

	return db, nil
}
```

3. To get email working you must provide your SMTP settings in the [config.yml](https://github.com/ricoberger/go-vue-starter/blob/master/config.yml#L14) file. For gmail for example you can use something like this:

```yaml
mail:
  identity: ""
  username: myname@gmail.com
  password: mypassword
  host: smtp.gmail.com
  addr: smtp.gmail.com:587
  from: myname@gmail.com
```

**Comment by ricoberger** [comment]
I closed the issue with the merge of #8. If you have any further questions feel free to reopen the issue or to create a new one.
